agent/websockets/connection.go

```diff
@@ -38,12 +38,36 @@ type Connection struct {
 	serverMessages chan string
 }
 
+// This map defines the set of headers that should be stripped from the WS request, as they
+// are being verified by the websocket.Client.
+var stripHeaderNames = map[string]bool{
+	"Upgrade":                  true,
+	"Connection":               true,
+	"Sec-Websocket-Key":        true,
+	"Sec-Websocket-Version":    true,
+	"Sec-Websocket-Extensions": true,
+	"Sec-Websocket-Protocol":   true,
+}
+
+// stripWSHeader strips request headers that are not allowed by the websocket.Client library,
+// while preserving the rest.
+func stripWSHeader(header http.Header) http.Header {
+	result := http.Header{}
+	for k, vals := range header {
+		if !stripHeaderNames[k] {
+			for _, v := range vals {
+				result.Add(k, v)
+			}
```

Comment:
I *think* this can just be `result[k] = vals`, but I don't have a strong opinion on that.

Reply:
Fixed